src/Native/GirTestLib/girtest-error-tester.c

```diff
@@ -0,0 +1,99 @@
+#include "girtest-error-tester.h"
+
+/**
+ * GirTestErrorTester:
+ *
+ * Contains functions to test GError-based error handling.
+ */
+
+G_DEFINE_QUARK (girtest-error-tester-error-quark, girtest_error_tester_error)
+
+struct _GirTestErrorTester
+{
+    GObject parent_instance;
+};
+
+G_DEFINE_TYPE(GirTestErrorTester, girtest_error_tester, G_TYPE_OBJECT)
+
+static void
+girtest_error_tester_init(GirTestErrorTester *value)
+{
+}
+
+static void
+girtest_error_tester_class_init(GirTestErrorTesterClass *class)
+{
+}
+
+/**
+ * girtest_error_tester_new_can_fail:
+ * @fail: Specifies whether the constructor should fail.
+ * @error: Return location for a potential error.
+ *
+ * Creates a new `GirTestErrorTester`, or produces an error depending on the boolean parameter.
+ *
+ * Returns: (transfer full) (nullable): The newly created `GirTestErrorTester`, on success.
+ */
+GirTestErrorTester*
+girtest_error_tester_new_can_fail (gboolean fail, GError **error)
+{
+    g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
```

Comment:
In [GdkPixbuf](https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/blob/master/gdk-pixbuf/gdk-pixbuf-io.c#L1119) this returns `null` which is better I think as you are returning `NULL` in the error case, too?

` g_return_val_if_fail (error == NULL || *error == NULL, NULL);`

Reply:
Good catch! That was a copy-paste error and only worked because of implicit conversions - agreed it should return `NULL`.